src/object/srv_obj_migrate.c

```diff
@@ -868,17 +870,12 @@ migrate_fetch_update_single(struct migrate_one *mrone, daos_handle_t oh,
 		}
 
 		if (obj_shard_is_ec_parity(mrone->mo_oid, NULL)) {
-			d_iov_t e_iov = { 0 };
-
 			rc = obj_ec_singv_encode_buf(mrone->mo_oid.id_pub,
+						     mrone->mo_oid.id_shard,
 						     iod, oca, &sgls[i],
-						     &e_iov);
+						     &sgls[i].sg_iovs[0]);
```

Comment:
well, maybe I did not explain my question clearly.
the issue is that here the returned data in sgls[i].sg_iovs[0] include all parity shards' data. here need to filter out this specific parity shard's data right?
for example 4+2 EC, when rebuilding shard 5 (the last parity shard). the above sgls[i].sg_iovs[0] with data of parity shard 4 and 5, here we need to filter out only shard 5's data out to write to VOS?
or maybe I misunderstood it...

Reply:
ok, i see, you changed obj_ec_singv_encode_buf().